gemfire-core/src/main/java/com/gemstone/gemfire/internal/LinuxProcFsStatistics.java

```diff
@@ -620,22 +620,47 @@ private static String getLineFromFile(String fileName) {
   }
 
   /**
+   * Get the current total number of threads across system and maximum number
+   * of file descriptors threads allowed by OS for the machine.
+   */
+  private static void getSystemFileDescriptorStats(long[] longs) {
+    String line = getLineFromFile("/proc/sys/fs/file-nr");
+    if (line != null) {
+      String[] parts = line.split("[ \t]");
+      if (parts.length > 0) {
+        try {
+          longs[LinuxSystemStats.totalFileDescriptorsLONG] =
+              Long.parseLong(parts[0]);
+        } catch (NumberFormatException ignored) {
+        }
+        if (parts.length > 2) {
+          try {
+            longs[LinuxSystemStats.fileDescriptorsSystemMaxLONG] =
+                Long.parseLong(parts[2]);
```

Comment:
The second field is number of allocated handles but unused. I don't think this is very useful for us in general.